[DEFECT] Unique field values failed to load under specific circumstances

### Problem Statement

An error is being thrown when dotCMS starts up in a customer instance. There's a specific use case that is not being considered, which must be handled correctly. The following error is being logged in the `dotcms.log` file:
```
2025-09-23T09:03:26-05:00	14:03:26.135  INFO  config.DotInitializationService - Initializing :com.dotcms.contenttype.business.uniquefields.UniqueFieldsValidationInitializer
2025-09-23T09:03:26-05:00	14:03:26.137  INFO  uniquefields.UniqueFieldsValidationInitializer - Creating and populating the Unique Fields table
2025-09-23T09:03:26-05:00	14:03:26.138  INFO  extratable.UniqueFieldDataBaseUtil - ---> Creating the unique_fields table
2025-09-23T09:03:26-05:00	14:03:26.145  INFO  extratable.UniqueFieldDataBaseUtil - ---> Populating the unique_fields table
2025-09-23T09:03:26-05:00	[   619.083606s]  WARN ThreadId(01) outbound:proxy{addr=52.6.71.83:8181}:forward{addr=52.6.71.83:8181}: linkerd_reconnect: Failed to connect error=connect timed out after 1s
2025-09-23T09:03:26-05:00	[   619.339980s]  INFO ThreadId(01) inbound: linkerd_app_core::serve: Connection closed error=connection closed before message completed client.addr=192.168.121.29:48854 server.addr=192.168.119.29:8082
2025-09-23T09:03:28-05:00	[   620.587285s]  WARN ThreadId(01) outbound:proxy{addr=52.6.71.83:8181}:forward{addr=52.6.71.83:8181}: linkerd_reconnect: Failed to connect error=connect timed out after 1s
2025-09-23T09:03:29-05:00	[   622.089290s]  WARN ThreadId(01) outbound:proxy{addr=52.6.71.83:8181}:forward{addr=52.6.71.83:8181}: linkerd_reconnect: Failed to connect error=connect timed out after 1s
2025-09-23T09:03:29-05:00	[   622.227425s]  INFO ThreadId(01) outbound:proxy{addr=52.6.71.83:8181}:rescue{client.addr=192.168.119.29:57072}: linkerd_app_core::errors::respond: gRPC request failed error=logical service 52.6.71.83:8181: route default.fallback: service unavailable error.sources=[route default.fallback: service unavailable, service unavailable]
2025-09-23T09:03:31-05:00	[   623.591905s]  WARN ThreadId(01) outbound:proxy{addr=52.6.71.83:8181}:forward{addr=52.6.71.83:8181}: linkerd_reconnect: Failed to connect error=connect timed out after 1s
2025-09-23T09:03:32-05:00	[   625.093788s]  WARN ThreadId(01) outbound:proxy{addr=52.6.71.83:8181}:forward{addr=52.6.71.83:8181}: linkerd_reconnect: Failed to connect error=connect timed out after 1s
2025-09-23T09:03:34-05:00	[   626.596424s]  WARN ThreadId(01) outbound:proxy{addr=52.6.71.83:8181}:forward{addr=52.6.71.83:8181}: linkerd_reconnect: Failed to connect error=connect timed out after 1s
2025-09-23T09:03:35-05:00	[   628.099280s]  WARN ThreadId(01) outbound:proxy{addr=52.6.71.83:8181}:forward{addr=52.6.71.83:8181}: linkerd_reconnect: Failed to connect error=connect timed out after 1s
2025-09-23T09:03:36-05:00	[   629.339343s]  INFO ThreadId(01) inbound: linkerd_app_core::serve: Connection closed error=connection closed before message completed client.addr=192.168.121.29:56908 server.addr=192.168.119.29:8082
2025-09-23T09:03:37-05:00	[   629.600772s]  WARN ThreadId(01) outbound:proxy{addr=52.6.71.83:8181}:forward{addr=52.6.71.83:8181}: linkerd_reconnect: Failed to connect error=connect timed out after 1s
2025-09-23T09:03:37-05:00	14:03:37.376  WARN  db.DotConnect - Somewhat slow query, total time: 11228.087ms, query preparation time: 0.009149ms, query execution time: 11228.076ms, metadata time: 0.0ms, SQL: INSERT INTO unique_fields (unique_key_val, supporting_values) SELECT  encode(            sha256(                    convert_to(                            CONCAT(                                    content_type_id::text,                                    field_var_name::text,                                    language_id::text,                                    LOWER(field_value)::text,                                    CASE WHEN uniquePerSite = 'true' THEN COALESCE(host_id::text, '') ELSE '' END                            ),                            'UTF8'                    )            ),            'hex'       ) AS unique_key_val,        json_build_object('contentTypeId', content_type_id, 'fieldVariableName', field_var_name, 'languageId', language_id, 'fieldValue', LOWER(field_value), 'siteId', host_id, 'variant', variant_id, 'uniquePerSite', uniquePerSite, 'live', live, 'contentletIds', contentlet_identifier) AS supporting_values FROM (        SELECT structure.inode                                       AS content_type_id,               field.velocity_var_name                               AS field_var_name,               contentlet.language_id                                AS language_id,               (CASE WHEN field_variable.variable_value = 'true' THEN identifier.host_inode ELSE '' END) AS host_id,               jsonb_extract_path_text(contentlet_as_json -> 'fields', field.velocity_var_name)::jsonb ->>'value' AS field_value,               ARRAY_AGG(DISTINCT contentlet.identifier)                      AS contentlet_identifier,               (CASE WHEN COUNT(DISTINCT contentlet_version_info.variant_id) > 1 THEN 'DEFAULT' ELSE MAX(contentlet_version_info.variant_id) END) AS variant_id,                ((CASE WHEN COUNT(*) > 1 AND COUNT(DISTINCT contentlet_version_info.live_inode = contentlet.inode) > 1 THEN 0                    ELSE MAX((CASE WHEN contentlet_version_info.live_inode = contentlet.inode THEN 1 ELSE 0 END)::int)                    END) = 1) AS live,               (MAX(CASE WHEN field_variable.variable_value = 'true' THEN 1 ELSE 0 END)) = 1 AS uniquePerSite        FROM contentlet                 INNER JOIN structure ON structure.inode = contentlet.structure_inode                 INNER JOIN field ON structure.inode = field.structure_inode                 INNER JOIN identifier ON contentlet.identifier = identifier.id                 INNER JOIN contentlet_version_info ON contentlet_version_info.live_inode = contentlet.inode OR                                                       contentlet_version_info.working_inode = contentlet.inode                 LEFT JOIN field_variable ON field_variable.field_id = field.inode AND field_variable.variable_key = 'uniquePerSite'        WHERE jsonb_extract_path_text(contentlet_as_json -> 'fields', field.velocity_var_name) IS NOT NULL          AND field.unique_ = true        GROUP BY structure.inode,                 field.velocity_var_name,                 contentlet.language_id,                 (CASE WHEN field_variable.variable_value = 'true' THEN identifier.host_inode ELSE '' END),                 jsonb_extract_path_text(contentlet_as_json -> 'fields', field.velocity_var_name)::jsonb ->>'value') as data_to_populate, parameters: []
2025-09-23T09:03:37-05:00	14:03:37.380  INFO  extratable.UniqueFieldDataBaseUtil - ---> Fixing and reporting records with the same hash, if any
2025-09-23T09:03:37-05:00	14:03:37.963  INFO  extratable.UniqueFieldDataBaseUtil - A total of 1 records with the same hash value were found
2025-09-23T09:03:37-05:00	14:03:37.967  INFO  extratable.UniqueFieldDataBaseUtil - Fixing conflict for Content IDs '[03486d28-3a83-4e14-91f3-418d910f86f8]' with duplicate unique value 'pa correlations k-5'
2025-09-23T09:03:37-05:00	14:03:37.998  INFO  extratable.UniqueFieldDataBaseUtil - Fixing conflict for Content IDs '[a20d8d31-27e2-4048-8959-c2244d213afd]' with duplicate unique value 'pa correlations k-5'
2025-09-23T09:03:38-05:00	14:03:38.021  INFO  extratable.UniqueFieldDataBaseUtil - ---> Bringing primary key constraints back
2025-09-23T09:03:38-05:00	14:03:38.123  WARN  struts.MultiMessageResources - Duplicate resource property definition (key=was ==> is now): contenttypes.content.copy=Copy ==> Copy
2025-09-23T09:03:38-05:00	14:03:38.125  WARN  struts.MultiMessageResources - Duplicate resource property definition (key=was ==> is now): message.category.import=Importing Categories.  This operation cannot be cancelled or undone. ==> IMPORT
2025-09-23T09:03:38-05:00	14:03:38.126  WARN  struts.MultiMessageResources - Duplicate resource property definition (key=was ==> is now): message.containers.create.content_type_code=Content Type Code ==> Content Type Code
2025-09-23T09:03:38-05:00	14:03:38.128  WARN  struts.MultiMessageResources - Duplicate resource property definition (key=was ==> is now): Save=Save ==> Save
2025-09-23T09:03:38-05:00	14:03:38.130  WARN  struts.MultiMessageResources - Duplicate resource property definition (key=was ==> is now): Move=Move ==> Move
2025-09-23T09:03:38-05:00	14:03:38.131  WARN  struts.MultiMessageResources - Duplicate resource property definition (key=was ==> is now): dot.common.remove=Remove ==> Remove
2025-09-23T09:03:38-05:00	14:03:38.131  WARN  struts.MultiMessageResources - Duplicate resource property definition (key=was ==> is now): archived=archived ==> Archived
2025-09-23T09:03:38-05:00	14:03:38.131  WARN  struts.MultiMessageResources - Duplicate resource property definition (key=was ==> is now): experiments.action.edit=Edit ==> Edit
2025-09-23T09:03:38-05:00	14:03:38.132  WARN  struts.MultiMessageResources - Duplicate resource property definition (key=was ==> is now): locales.edit.locale=Edit Locale ==> Edit Locale
2025-09-23T09:03:38-05:00	14:03:38.132  WARN  struts.MultiMessageResources - Duplicate resource property definition (key=was ==> is now): locales.add.locale=Add Locale ==> Add Locale
2025-09-23T09:03:38-05:00	14:03:38.132  WARN  struts.MultiMessageResources - Duplicate resource property definition (key=was ==> is now): locale.id.hint=Locale is determined by the Language used, or generated by combining Language and Country 2-letter ISO codes. ==> Locales should generally be concise. Many sites use ISO codes, such as 'en' for English or 'es-cr' for Español in Costa Rica.
2025-09-23T09:03:38-05:00	14:03:38.507  ERROR extratable.UniqueFieldDataBaseUtil - Failed to bring primary key constraints back. There may be an unhandled unique value scenario
2025-09-23T09:03:38-05:00	14:03:38.515  WARN  uniquefields.UniqueFieldsValidationInitializer - ERROR: could not create unique index "unique_fields_pkey"
2025-09-23T09:03:38-05:00	  Detail: Key (unique_key_val)=(367867566ab2dfa900c0a5b6076321b07f9260bf3627bda7a182436167b8439a) is duplicated.{
2025-09-23T09:03:38-05:00	  "SQL": ["ALTER TABLE unique_fields ADD PRIMARY KEY (unique_key_val)"],
2025-09-23T09:03:38-05:00	  "maxRows": [-1],
2025-09-23T09:03:38-05:00	  "offest": [0]
2025-09-23T09:03:38-05:00	}
2025-09-23T09:03:38-05:00	14:03:38.515  WARN  uniquefields.UniqueFieldsValidationInitializer - com.dotmarketing.common.db.DotConnect.loadResult(DotConnect.java:336)
2025-09-23T09:03:38-05:00	14:03:38.516  WARN  uniquefields.UniqueFieldsValidationInitializer - Dropping 'unique_fields' table so that the process can run again the next time
2025-09-23T09:03:38-05:00	14:03:38.544  ERROR config.DotInitializationService - ERROR: could not create unique index "unique_fields_pkey"
2025-09-23T09:03:38-05:00	  Detail: Key (unique_key_val)=(367867566ab2dfa900c0a5b6076321b07f9260bf3627bda7a182436167b8439a) is duplicated.{
2025-09-23T09:03:38-05:00	  "SQL": ["ALTER TABLE unique_fields ADD PRIMARY KEY (unique_key_val)"],
2025-09-23T09:03:38-05:00	  "maxRows": [-1],
2025-09-23T09:03:38-05:00	  "offest": [0]
2025-09-23T09:03:38-05:00	}
2025-09-23T09:03:38-05:00	com.dotmarketing.exception.DotRuntimeException: ERROR: could not create unique index "unique_fields_pkey"
2025-09-23T09:03:38-05:00	  Detail: Key (unique_key_val)=(367867566ab2dfa900c0a5b6076321b07f9260bf3627bda7a182436167b8439a) is duplicated.{
2025-09-23T09:03:38-05:00	  "SQL": ["ALTER TABLE unique_fields ADD PRIMARY KEY (unique_key_val)"],
2025-09-23T09:03:38-05:00	  "maxRows": [-1],
2025-09-23T09:03:38-05:00	  "offest": [0]
2025-09-23T09:03:38-05:00	}
2025-09-23T09:03:38-05:00		at com.dotcms.contenttype.business.uniquefields.UniqueFieldsValidationInitializer.init(UniqueFieldsValidationInitializer.java:66) ~[?:?]
2025-09-23T09:03:38-05:00		at com.dotcms.config.DotInitializationService.initService(DotInitializationService.java:76) ~[?:?]
2025-09-23T09:03:38-05:00		at java.base/java.lang.Iterable.forEach(Iterable.java:75) ~[?:?]
2025-09-23T09:03:38-05:00		at com.dotcms.config.DotInitializationService.initialize(DotInitializationService.java:116) ~[?:?]
2025-09-23T09:03:38-05:00		at com.liferay.portal.servlet.MainServlet.init(MainServlet.java:211) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.core.StandardWrapper.initServlet(StandardWrapper.java:989) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.core.StandardWrapper.loadServlet(StandardWrapper.java:946) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.core.StandardWrapper.load(StandardWrapper.java:843) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.core.StandardContext.loadOnStartup(StandardContext.java:4234) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:4535) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:164) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:599) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:571) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:603) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.startup.HostConfig.deployDirectory(HostConfig.java:1174) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.startup.HostConfig$DeployDirectory.run(HostConfig.java:1883) ~[?:?]
2025-09-23T09:03:38-05:00		at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572) ~[?:?]
2025-09-23T09:03:38-05:00		at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:76) ~[?:?]
2025-09-23T09:03:38-05:00		at java.base/java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:123) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.startup.HostConfig.deployDirectories(HostConfig.java:1085) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:470) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1579) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:312) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:109) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.util.LifecycleBase.setStateInternal(LifecycleBase.java:389) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.util.LifecycleBase.setState(LifecycleBase.java:336) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:776) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:721) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:164) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1203) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1193) ~[?:?]
2025-09-23T09:03:38-05:00		at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:76) ~[?:?]
2025-09-23T09:03:38-05:00		at java.base/java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:145) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:749) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:211) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:164) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.core.StandardService.startInternal(StandardService.java:412) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:164) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:874) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:164) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.startup.Catalina.start(Catalina.java:739) ~[?:?]
2025-09-23T09:03:38-05:00		at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?]
2025-09-23T09:03:38-05:00		at java.base/java.lang.reflect.Method.invoke(Method.java:580) ~[?:?]
2025-09-23T09:03:38-05:00		at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:345) ~[bootstrap.jar:9.0.108]
2025-09-23T09:03:38-05:00		at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:476) ~[bootstrap.jar:9.0.108]
2025-09-23T09:03:38-05:00	Caused by: com.dotmarketing.exception.DotDataException: ERROR: could not create unique index "unique_fields_pkey"
2025-09-23T09:03:38-05:00	  Detail: Key (unique_key_val)=(367867566ab2dfa900c0a5b6076321b07f9260bf3627bda7a182436167b8439a) is duplicated.{
2025-09-23T09:03:38-05:00	  "SQL": ["ALTER TABLE unique_fields ADD PRIMARY KEY (unique_key_val)"],
2025-09-23T09:03:38-05:00	  "maxRows": [-1],
2025-09-23T09:03:38-05:00	  "offest": [0]
2025-09-23T09:03:38-05:00	}
2025-09-23T09:03:38-05:00		at com.dotmarketing.common.db.DotConnect.loadResult(DotConnect.java:336) ~[?:?]
2025-09-23T09:03:38-05:00		at com.dotmarketing.common.db.DotConnect.loadObjectResults(DotConnect.java:1023) ~[?:?]
2025-09-23T09:03:38-05:00		at com.dotcms.contenttype.business.uniquefields.extratable.UniqueFieldDataBaseUtil.addPrimaryKeyConstraintsBack(UniqueFieldDataBaseUtil.java:510) ~[?:?]
2025-09-23T09:03:38-05:00		at com.dotcms.contenttype.business.uniquefields.extratable.UniqueFieldDataBaseUtil.createTableAndPopulate(UniqueFieldDataBaseUtil.java:426) ~[?:?]
2025-09-23T09:03:38-05:00		at com.dotcms.contenttype.business.uniquefields.extratable.UniqueFieldDataBaseUtil$Proxy$_$$_WeldClientProxy.createTableAndPopulate(Unknown Source) ~[?:?]
2025-09-23T09:03:38-05:00		at com.dotcms.contenttype.business.uniquefields.UniqueFieldsValidationInitializer.init(UniqueFieldsValidationInitializer.java:53) ~[?:?]
2025-09-23T09:03:38-05:00		... 46 more
2025-09-23T09:03:38-05:00	Caused by: org.postgresql.util.PSQLException: ERROR: could not create unique index "unique_fields_pkey"
2025-09-23T09:03:38-05:00	  Detail: Key (unique_key_val)=(367867566ab2dfa900c0a5b6076321b07f9260bf3627bda7a182436167b8439a) is duplicated.
2025-09-23T09:03:38-05:00		at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2725) ~[?:?]
2025-09-23T09:03:38-05:00		at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2412) ~[?:?]
2025-09-23T09:03:38-05:00		at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:371) ~[?:?]
2025-09-23T09:03:38-05:00		at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:502) ~[?:?]
2025-09-23T09:03:38-05:00		at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:419) ~[?:?]
2025-09-23T09:03:38-05:00		at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:341) ~[?:?]
2025-09-23T09:03:38-05:00		at org.postgresql.jdbc.PgStatement.executeCachedSql(PgStatement.java:326) ~[?:?]
2025-09-23T09:03:38-05:00		at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:302) ~[?:?]
2025-09-23T09:03:38-05:00		at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:297) ~[?:?]
2025-09-23T09:03:38-05:00		at com.zaxxer.hikari.pool.ProxyStatement.execute(ProxyStatement.java:95) ~[?:?]
2025-09-23T09:03:38-05:00		at com.zaxxer.hikari.pool.HikariProxyStatement.execute(HikariProxyStatement.java) ~[?:?]
2025-09-23T09:03:38-05:00		at com.dotmarketing.common.db.DotConnect.executeQuery(DotConnect.java:724) ~[?:?]
2025-09-23T09:03:38-05:00		at com.dotmarketing.common.db.DotConnect.executeQuery(DotConnect.java:626) ~[?:?]
2025-09-23T09:03:38-05:00		at com.dotmarketing.common.db.DotConnect.loadResult(DotConnect.java:334) ~[?:?]
2025-09-23T09:03:38-05:00		at com.dotmarketing.common.db.DotConnect.loadObjectResults(DotConnect.java:1023) ~[?:?]
2025-09-23T09:03:38-05:00		at com.dotcms.contenttype.business.uniquefields.extratable.UniqueFieldDataBaseUtil.addPrimaryKeyConstraintsBack(UniqueFieldDataBaseUtil.java:510) ~[?:?]
2025-09-23T09:03:38-05:00		at com.dotcms.contenttype.business.uniquefields.extratable.UniqueFieldDataBaseUtil.createTableAndPopulate(UniqueFieldDataBaseUtil.java:426) ~[?:?]
2025-09-23T09:03:38-05:00		at com.dotcms.contenttype.business.uniquefields.extratable.UniqueFieldDataBaseUtil$Proxy$_$$_WeldClientProxy.createTableAndPopulate(Unknown Source) ~[?:?]
2025-09-23T09:03:38-05:00		at com.dotcms.contenttype.business.uniquefields.UniqueFieldsValidationInitializer.init(UniqueFieldsValidationInitializer.java:53) ~[?:?]
2025-09-23T09:03:38-05:00		... 46 more
...
...
...
```

### Steps to Reproduce

Just start up a dotCMS instance with the latest main, using a specific database. Please reach out to Jose Castro for more info about it.

### Acceptance Criteria

- [ ] dotCMS must start up correctly.
- [ ] For the conflicting Contentlets: One must be editable correctly (as the source of truth), and the other ones must fail as they're using the already existing unique value.
- [ ] Create an Integration Test for the new use case.

### dotCMS Version

25.09.22-1, and latest main.

### Severity

Critical - System unusable

--- a/dotCMS/src/main/java/com/dotcms/contenttype/business/uniquefields/extratable/UniqueFieldDataBaseUtil.java
+++ b/dotCMS/src/main/java/com/dotcms/contenttype/business/uniquefields/extratable/UniqueFieldDataBaseUtil.java
@@ -35,6 +35,7 @@ import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
+import java.util.Random;
 
 import static com.dotcms.contenttype.business.uniquefields.extratable.SqlQueries.DELETE_UNIQUE_FIELDS;
 import static com.dotcms.contenttype.business.uniquefields.extratable.SqlQueries.DELETE_UNIQUE_FIELDS_BY_CONTENTLET;
@@ -689,14 +690,22 @@ public class UniqueFieldDataBaseUtil {
 
     /**
      * Generates a new unique value based on the original one, using a specific format that can also
-     * be used to look for conflicting entries that were fixed.
+     * be used to look for conflicting entries that were fixed. This new value is composed of:
+     * <ol>
+     *     <li>The {@code legacy*support*} prefix.</li>
+     *     <li>A random number between 1 and 100,000, which is required in case there are more than
+     *     two Contentlets with the exact same unique value so that values don't conflict with each
+     *     other.</li>
+     *     <li>The original unique value enclosed in curly braces.</li>
+     * </ol>
      *
      * @param originalUniqueValue The original unique value.
      *
      * @return The new unique value.
      */
     private String generateUniqueName(final String originalUniqueValue) {
-        return "legacy*support*{" + originalUniqueValue + "}";
+        int randomNumber = new Random().nextInt(100000) + 1;
+        return "legacy*support*" + randomNumber + "{" + originalUniqueValue + "}";
     }
 
     /**

--- a/dotcms-integration/src/test/java/com/dotcms/contenttype/business/uniquefields/extratable/UniqueFieldDataBaseUtilTest.java
+++ b/dotcms-integration/src/test/java/com/dotcms/contenttype/business/uniquefields/extratable/UniqueFieldDataBaseUtilTest.java
@@ -1189,8 +1189,10 @@ public class UniqueFieldDataBaseUtilTest {
     /**
      * <ul>
      *     <li><b>Method to test:</b> {@link UniqueFieldDataBaseUtil#handleDuplicateRecords()}</li>
-     *     <li><b>Given Scenario:</b> Manually insert two records in the {@code unique_fields} table
-     *     having the exact same hash but point to different Contentlet Identifiers.</li>
+     *     <li><b>Given Scenario:</b> Manually insert four records in the {@code unique_fields}
+     *     table having the exact same hash, but point to different Contentlet Identifiers. This
+     *     simulates the scenario in which different contents have the same unique value, which
+     *     should NEVER happen but was being allowed by race conditions with Elasticsearch.</li>
      *     <li><b>Expected Result:</b> The method to test will detect the duplicates and manually
      *     fixes the conflicts. It leaves only one of the duplicates, updates the unique value of
      *     the remaining ones, and re-generates their hashes so they can be inserted in the table
@@ -1211,6 +1213,8 @@ public class UniqueFieldDataBaseUtilTest {
         final String siteId = UUIDGenerator.generateUuid();
         final String testContentIdOne = UUIDGenerator.generateUuid();
         final String testContentIdTwo = UUIDGenerator.generateUuid();
+        final String testContentIdThree = UUIDGenerator.generateUuid();
+        final String testContentIdFour = UUIDGenerator.generateUuid();
         final long languageId = 1;
 
         // ╔════════════════════════╗
@@ -1255,15 +1259,21 @@ public class UniqueFieldDataBaseUtilTest {
             ));
             uniqueFieldDataBaseUtil.insert(uniqueKeyValue, supportingValues);
 
-            // Generating the duplicate record for the second Contentlet
+            // Generating the duplicate unique value record for the second Contentlet
             supportingValues.put(CONTENTLET_IDS_ATTR, List.of(testContentIdTwo));
+            uniqueFieldDataBaseUtil.insert(uniqueKeyValue, supportingValues);
+            // Generating the duplicate unique value record for the third Contentlet
+            supportingValues.put(CONTENTLET_IDS_ATTR, List.of(testContentIdThree));
+            uniqueFieldDataBaseUtil.insert(uniqueKeyValue, supportingValues);
+            // Generating the duplicate unique value record for the fourth Contentlet
+            supportingValues.put(CONTENTLET_IDS_ATTR, List.of(testContentIdFour));
             uniqueFieldDataBaseUtil.insert(uniqueKeyValue, supportingValues);
 
             // ╔══════════════╗
             // ║  Assertions  ║
             // ╚══════════════╝
             List<Map<String, Object>> recordCountWithSameHash = this.getUniqueFieldRecordsWithSameHash();
-            assertEquals("There must be exactly 2 duplicate records", 2,
+            assertEquals("There must be exactly 4 duplicate records", 4,
                     Integer.parseInt(recordCountWithSameHash.get(0).get("count").toString()));
 
             uniqueFieldDataBaseUtil.handleDuplicateRecords();

